src/docs/message-handling.dox

```diff
@@ -0,0 +1,100 @@
+application-specific logging function that added a timestamp and logged
+the message to a file, and error messages might additionally be output to
+the \c stderr file stream.
+
+Additionally, applications will normally handle \c WT_PANIC as a special
+case. WiredTiger will always call the error handler callback with
+\c WT_PANIC in the case of a fatal error requiring database restart,
+however, WiredTiger cannot guarantee applications will see an application
+thread return \c WT_PANIC from a WiredTiger API call. For this reason, a
+correctly-written WiredTiger application will likely specify at least an
+error handler which will immediately exit or otherwise handle fatal errors.
+Note that no further WiredTiger calls are required after an error handler
+is called with \c WT_PANIC (and further calls will themselves immediately
+fail).
+
+The following is a programmatic example of creating a custom event handler:
+
+@snippet ex_event_handler.c Function event_handler
+
+The following is a programmatic example of configuring the previously defined custom event handler:
+
+@snippet ex_event_handler.c Configure event_handler
+
+@section message_handling_output Output format of the WT_EVENT_HANDLER
+
+The messages generated by the WT_EVENT_HANDLER can follow a flat string style format or can be
+encoded in the JSON format.
+
+@subsection message_handling_output_flat_string Flat string format
+
+By default, all the messages going through the WT_EVENT_HANDLER are generated as flat strings.
+Example of a message following the flat string format:
+
+@code
+[1637732238:116292][11175:0x7fd106107740], test_salvage.test_salvage.test_salvage_api(column-fix), file:test_salvage.a.wt, WT_SESSION.salvage: [WT_VERB_SALVAGE_PROGRESS][INFO]: WT_SESSION.salvage
+@endcode
+
+@subsection message_handling_output_json JSON format
+
+The connection setting \c json_ouput can be configured to enable JSON encoding for specific message
+categories. For instance, to print all \b general and \b error messages in JSON format, the
+following setting needs to be added to the connection configuration:
+
+@code
+json_output=(message, error)
+@endcode
+
+Any type of message that is not specified in the list assigned to the \c json_output setting will be
+generated following the flat string format. When the json_output setting is not specified, all
+messages are generated as flat strings.
+
+The JSON output follows a specific schema with the following fields:
+
+<table>
+@hrow{Field, Description}
+@row{category, Message category defined by \c WT_VERBOSE_CATEGORY}
+@row{category_id, Value representation of the message category}
+@row{error_code, Value representation of the error}
+@row{error_str, Error message}
+@row{msg, Message}
+@row{session_dhandle_name, Session dhandle name}
+@row{session_err_prefix, Database error prefix defined in the \c WT_CONNECTION_IMPL structure}
+@row{session_name, Session name}
+@row{ts_sec, Time in seconds at which the message was generated}
```

Comment:
Might want to be specific about the time measurement used here. 
It should be based on the system-wide real clock time since Epoch

Reply:
Updated in 3ab02be8ede943f08d2caefdfa0049eac7281772